libbpf-tools/klockstat.c

```diff
@@ -549,6 +549,26 @@ int main(int argc, char **argv)
 					       "mutex_lock_killable_nested");
 	}
 
+	if (fentry_can_attach("down_read_nested", NULL)) {
```

Comment:
this nested stuff is a little worrisome.  i see you're doing the same thing as the code did for mutex_lock, but i'm not familiar with that change either.  are we tracing actual nested locks, or is that some kprobe-naming issue?

klockstat.bpf has an assumption that a given task won't try to grab the same lock more than once at a time.  if it does, you'll get corrupted results.

that's not a reason to block the rwsem change, but it is a concern with any use of nested locks (mutex or otherwise).

Reply:
I believe even for the nested lock paths, they will see different lock instances at each invocation.  It's about the lock classes (in lockdep) rather than instances.

And yeah, it's more about the kprobe-naming issue.  When CONFIG_DEBUG_LOCK_ALLOC is defined, the regular `mutex_lock` will be renamed to `mutex_lock_nested`.  On the other hand, if the config option is not selected, `mutex_lock_nested` will be renamed to `mutex_lock`.